source/digits_hits/include/GateJPETGammaPolarizationActor.hh

```diff
@@ -0,0 +1,302 @@
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  @file GateJPETGammaPolarizationActor.hh
+ */
+
+#include "GateConfiguration.h"
+
+#ifndef GATEJPETGAMMAPOLARIZATIONACTOR_HH
+#define GATEJPETGAMMAPOLARIZATIONACTOR_HH
+
+#include "TROOT.h"
+#include "TFile.h"
+#include "TNtuple.h"
+#include "TTree.h"
+#include "TBranch.h"
+#include "TString.h"
+#include<fstream>
+#include "TH2.h"
+#include "TH1.h"
+#include<map>
+
+#include "GateVActor.hh"
+#include "GateJPETGammaPolarizationActorMessenger.hh"
+
+struct iaea_header_type;
+struct iaea_record_type;
+
+
+/**
+ * Actor created just for gamma quanta's polarization analysis in Compton process (for JPET use)
+ * @author: Mateusz Bała
+ * @email: bala.mateusz@gmail.com
+ */
+class GateJPETGammaPolarizationActor : public GateVActor
+{
+public: //Functions, structures and enums
+	/** Simple structure which represent branch with test for polarization.
+	 * */
+	struct PolarizationControlTests{
+		//Angle on linear polarization on plane orthogonal to gamma direction
+		G4double ScatteredGammaPolarizationAngle;
+		//Angular difference between polarization received from simulation and as a cross product of prime and scattered gamma (JPET idea)
+		G4double AngularDiffreneceBetweenRealAndReconstructedPrimeGammaPolarization;
+	};
+
+	/** This structure contain all information about track before and after scattering.
+	 * The purpose of this is to provide simple way to save data only when is needed to files
+	 * */
+	struct EventTrackPartner{
+		G4bool ComptonHappened;
+
+		//Information about particle before Compton's scattering
+		G4ThreeVector PrimeGammaPolarization;
+		G4ThreeVector PrimeGammaDirection;
+		G4double PrimeTotalEnergy;
+		G4double PrimePolarizationAngle;
+
+		//Information about particle when was emitted from source
+		G4ThreeVector PrimeEmissionPoint;
+		G4double PrimeEmisionPhi;
+		G4double PrimeEmisionTheta;
+		G4double PrimeEmissionKineticEnergy;
+
+		//Information about particle after Compton's scattering
+		G4double ComptonPhi;
+		G4double ComptonTheta;
+		G4ThreeVector ComptonInteractionPoint;
+		G4double ComptonTotatEnergy;
+		PolarizationControlTests Tests;
+	};
+
+	enum PhiThetaValueMode{
+		Flag,
+		Limes,
+		Epsilon
+	};
+
+	virtual ~GateJPETGammaPolarizationActor();
+	FCT_FOR_AUTO_CREATOR_ACTOR(GateJPETGammaPolarizationActor)
+	// Constructs the sensor
+	virtual void Construct();
+	// Callbacks
+	virtual void UserSteppingAction(const GateVVolume *, const G4Step*);
```

Comment:
not necessary but if could also use override keyword for the methods that are overridden